HW2/src/test/java/com/example/hw2/servise/AuthorServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package com.example.hw2.servise;
+
+import com.example.hw2.entity.Author;
+import com.example.hw2.repository.AuthorRepository;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class AuthorServiceTest {
+
+    @Mock
+    private AuthorRepository repository;
+    @InjectMocks
+    private AuthorService service;
```

Comment:
Prefer do not use `@InjectMocks`.

Use explicit instantiation in `@BeforeEach` instead.

It will show you all places that will be affected by your change and it will force you to make this change (if you run locally, the app will crush and you have to fix the test - more likely you will not push broken test in the repo).

It's kind of "Remainder" for you, that you need to add/fix some tests, and for reviewer, that test class affected but no tests added/changed there.